### DISCUSSION TITLE
How to query range date between startField and endField

### DISCUSSION THREAD

**Discussion by triasbrata** [prisma/prisma#22023] [Q&A]
### Question

how to create query like this

SELECT * FROM foo f WHERE '2023-10-10 10:00:00' between f.start_date AND f.end_date

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @triasbrata 👋 

Thank you for raising this question.

You are looking for queries that filter according to "date ranges". Considering you have an `Event` model that would look something like this:
```prisma
model Event {
  id        Int      @id @default(autoincrement())
  name      String
  startDate DateTime
  endDate   DateTime
}
```
Your query would look something like this:
```js
const searchDate = new Date('2023-06-07'); // replace with your search date

const events = await prisma.event.findMany({
  where: {
    startDate: {
      lte: searchDate,
    },
    endDate: {
      gte: searchDate, 
    },
  },
});
```
You can take a look at this [section](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#lt) of the documentation.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏
